### DISCUSSION TITLE
Is this ShazamIO "legal" for commercial use?

### DISCUSSION THREAD

**Discussion by nicolad76** [shazamio/ShazamIO#128] [Q&A]
Dears,
as much as I love this "piece of code" :) I cannot help but having a question: the license is valid for commercial use but the code itself is based on reverse engineered code from Shazam (read Apple). I went through the license from Apple and it states that it is forbidden to reverse engineer their code. Can someone explain to me if this code is violating Shazam license?

**Comment by dotX12** [comment]
> Dears, as much as I love this "piece of code" :) I cannot help but having a question: the license is valid for commercial use but the code itself is based on reverse engineered code from Shazam (read Apple). I went through the license from Apple and it states that it is forbidden to reverse engineer their code. Can someone explain to me if this code is violating Shazam license?

Hi, I can't say anything about it, the code is provided as is to study the algorithms of how the song search works and so on, any commercial use is at your own risk, I can't be held responsible for it like their license :))))))

**Comment by dotX12** [comment]
In fact no one provides a legal API for shazam, all API providers like RAPID API are in fact also violating the Shazam/Apple license.
